tx.go

```diff
@@ -91,12 +91,20 @@ func NewBatchTransaction(creator *skademlia.Keypair, tags []byte, payloads [][]b
 	return NewTransaction(creator, sys.TagBatch, append([]byte{byte(len(tags))}, buf...))
 }
 
-func AttachSenderToTransaction(sender *skademlia.Keypair, tx Transaction, parents ...*Transaction) Transaction {
+func AttachSenderToTransaction(sender *skademlia.Keypair, _tx *Transaction, parents ...*Transaction) *Transaction {
+	tx := *_tx
```

Comment:
why creating new transaction, sending it by pointer and returning new pointer? since this function already takes pointer its kinda obvious that it will be mutated.
maybe we should follow on of the two:
- either just mutate object coming from parameters and do not return it
- return value from function (sender and parents in this context) and attach them to transaction on upper level

current version is much more confusing than with no new structure returned
what are we caring about - less allocation/speed or immutability/safety(arguably)? lets decide one style and follow it.

Reply:
I agree as well. I would prefer  we care more about immutability/safety to be sure that the code itself is self-describing, much like idiomatic Go in Go's standard library.